cmp/compare_test.go

```diff
@@ -308,9 +335,11 @@ func comparerTests() []test {
 			return &c
 		}(),
 		wantDiff: `
-***{***int}:
-	-: 0
-	+: 1`,
+  &&&int(
```

Comment:
At first glance, I'd expect that this should be ***int instead since the comparison is on the dereferenced value.  And the example that uses intPtr() should be &int instead of *int.

But then I realized on types like map[*X]Y, the diff shows the key with pointer value instead of the dereferenced value and also it is not syntactically correct to have map[&X]Y.  But perhaps this is an exception, similar to []*T, map[S]*T, but the actual value for *T here would have &T{ ... } anyways unless it is a map key.

So, I think I have a minor preference for ***int on this one.

Reply:
In Go syntax, * means dereference, while & means to take the address of.

In the previous reporter output, the path was an expression from the root variable to the specific value. In that situation, you want to dereference pointers (hence the * operator).

In the new reporter output, we're constructing a pseudo Go-literal. There is no existing value to de-reference. Instead we are taking the address of the inner-most literal.